src/rules/no-unused-modules.js

```diff
@@ -201,6 +203,49 @@ const newNamespaceImportExists = specifiers =>
 const newDefaultImportExists = specifiers =>
   specifiers.some(({ type }) => type === IMPORT_DEFAULT_SPECIFIER)
 
+const fileIsInPkg = file => {
+  const { path, pkg } = readPkgUp.sync({cwd: file, normalize: false})
+  const basePath = dirname(path)
+
+  if (pkg.private === true) {
+    return false
+  }
+
+  if (pkg.bin && typeof pkg.bin === 'string') {
```

Comment:
seems like we can consolidate the logic for “check string field X” and “check object field Y”, so t doesn’t have to be repeated for main/bin/browser and bin/browser?

Reply:
Good hint. I mad the logic a bit more generic with 3ed222d.